PROTO_tests/tests/join_test.py

```diff
@@ -0,0 +1,209 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+"""
+Encapsulates a variety of arkouda join_on_eq_with_dt test cases.
+"""
+
+
+class TestJoin:
+    @classmethod
+    def setup_class(cls):
+        cls.size = 1000
+        cls.a1 = ak.ones(cls.size, dtype=np.int64)
+        cls.a2 = ak.arange(0, cls.size, 1)
```

Comment:
we don't have to do the first and last arguments here (stop = 0 and stride =1 by default)
```suggestion
        cls.a2 = ak.arange(cls.size)
```